pkg/scheduler/scheduler.go

```diff
@@ -282,10 +282,11 @@ func (sched *Scheduler) recordSchedulingFailure(pod *v1.Pod, err error, reason s
 	sched.config.Error(pod, err)
 	sched.config.Recorder.Event(pod, v1.EventTypeWarning, "FailedScheduling", message)
 	sched.config.PodConditionUpdater.Update(pod, &v1.PodCondition{
-		Type:    v1.PodScheduled,
-		Status:  v1.ConditionFalse,
-		Reason:  reason,
-		Message: err.Error(),
+		Type:          v1.PodScheduled,
+		Status:        v1.ConditionFalse,
+		LastProbeTime: metav1.Now(),
```

Comment:
Since the active queue takes into account lastProbeTime, and if a pod keeps getting unschedulable , which means its lastProbeTime will always be latest, it will be be farthest behind in the queue and hence least likely to be considered for scheduling ?  Is that the general idea ?

After being marked as unschedulable, how often is it tried again ? 

If an unschedulable pod eventually becomes schedulable, the LastTransitionTime will still update for this condition , hence this will still be least in prioirty to get schedulable compared to other pods. I am wondering if that will cause starvation ?

Reply:
>  Is that the general idea ?

Thank you for your clear explanation 🙇  Yes, it is.  That's my idea.

> After being marked as unschedulable, how often is it tried again ?

I think it depends on cluster status.  `MoveAllToActiveQueue()` moves unschedulable pods from unschedulable queue to active queue with backoff per pod.

> If an unschedulable pod eventually becomes schedulable, the LastTransitionTime will still update for this condition

[`LastTransitionTime` ](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions) should be updated only when the condition status is changed by definition.  This means, once the pod is marked as unschedulable, `LastTransitionTime` shoudn't be updated until condition status will become `True`.  It is because I added code updating `LastProbeTime` when a schedule failed.

I think`kubelet` is responsible for updating `PodScheduled.Status = True` of `PodScheduled` condition in the current implementation.

Reply:
What  I was trying to say is that this change optimizes for the cases where there are lot of unschedulable pods and favors other pods scheduling in that case.  Does it make the recovery of  a pod which has been unschedulable for a while and just became schedulable slower compare to previously @everpeace @bsalamat  because its been keep pushing to end because of constant updating of lastprobetime ?

Reply:
@krmayankk I don't think so. If an unschedulable pod has higher priority, it will still get to the head of the queue even after this change. When it has the same priority as other pods, it is fair to put it behind other pods with the same priority after the scheduler has tried it and determined that it is unschedulable.

Reply:
@bsalamat  i was talking about the case when there is no priority involved. All pods are same priroity or default priority. In that case its trying to avoid starvation for regular pods when lot of unschedulable pods are present. How does it affect the recovery of the unschedulable pods which finally become shcedulable. Does this behavior change when compared to without this change ? 
Note: Just trying to understand, the answer may be no change. It depends on how the active queue is implemented

Reply:
With this change (and somewhat similarly after #71488), a pod that is determined unschedulable goes behind other similar priority pods in the scheduling queue. Once pods become schedulable they are processed by their order in the scheduling queue. So, depending on their location in the queue, they may get scheduled before or after same priority pods.
In short, we don't expect further delays in scheduling unschedulable pods after this change.